OpenUpSA/wazimap-ng-ui
# async point counts

## Description
* Getting the point counts async after the facilities are created. 
   - Facilities are created with a loading spinner instead of facility count 
   - The facility count API is called(this uses dummy data for now until the backend is ready)
   - Facility counts are populated and the loading spinners are removed

* Loading state while the facilities are getting created
   - When a new geo is clicked, facilities go into the loading state immediately
   - `Loading` text is shown instead of facilities description
   - `Loading` button is shown instead of Show/Hide facilities button
   - `Download all data` button is disabled
   - Facilities pane is hidden
   - This is how it looks when it goes into the loading state
![image](https://user-images.githubusercontent.com/53019884/134784257-20d51a52-1285-4bc1-b2e5-9fe901a5cce6.png)

* Refactored the facilities code too so the code is simpler to read. Before we used `src/js/profile/profile_header.js` to create the facilities but now there is `facilities` folder in `profile` folder. 

![image](https://user-images.githubusercontent.com/53019884/134784090-5a57712b-5aff-49d3-a782-407b7a6ccc12.png)

In the image above
a - `src/js/profile/profile_header.js` manages breadcrumbs and geo description
b - `src/js/profile/facilities/facility_controller.js` manages the facility wrapper, facilities description, downloading all data, expanding / hiding the facilities etc
c - `src/js/profile/facilities/facility.js` manages the facility cards
d - `src/js/profile/facilities/facilityItem.js` manages each row of facility cards


## Related Issue
https://wazimap.atlassian.net/browse/WNCM-19

## How is it tested automatically?
* A GUI test is created `__tests__/gui/facilities.feature`
* Tests if the facilities are created correctly
* Tests if it goes into the loading state when the map is navigating to a new geo

## How should a reviewer test it locally
* Expand the rich data panel
* Confirm that the facilities are created correctly
* Confirm that the downloads are working
* Confirm that when navigating to a new geo, facilities pane goes into the loading state correctly


## Screenshots


## Changelog

### Added
* `src/js/profile/facilities/facility_controller.js` manages the facility wrapper, facilities description, downloading all data, expanding / hiding the facilities etc
* `src/js/profile/facilities/facility.js` manages the facility cards
* `src/js/profile/facilities/facilityItem.js` manages each row of facility cards
*  `__tests__/gui/facilities.feature` GUI test

### Updated
* `src/js/profile/profile_header.js` removed the code that is related with facilities

### Removed


## Checklist


### Pull Request


### Code Quality


### Testing
